src/main/java/com/kobylynskyi/graphql/codegen/GraphqlDocumentParser.java

```diff
@@ -2,17 +2,33 @@
 
 import com.kobylynskyi.graphql.codegen.utils.Utils;
 import graphql.language.Document;
+import graphql.parser.MultiSourceReader;
 import graphql.parser.Parser;
 
 import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
 
 class GraphqlDocumentParser {
 
     private static final Parser GRAPHQL_PARSER = new Parser();
 
     static Document getDocument(String schemaFilePath) throws IOException {
-        String fileContent = Utils.getFileContent(schemaFilePath);
-        return GRAPHQL_PARSER.parseDocument(fileContent);
+        return getDocument(Collections.singletonList(schemaFilePath));
     }
 
+    static Document getDocument(List<String> schemaPaths) throws IOException {
+        MultiSourceReader reader = createMultiSourceReader(schemaPaths);
+        return GRAPHQL_PARSER.parseDocument(reader);
+    }
+
+    private static MultiSourceReader createMultiSourceReader(List<String> schemaPaths) throws IOException {
+        MultiSourceReader.Builder builder = MultiSourceReader.newMultiSourceReader();
+        for (String path : schemaPaths) {
+            // appending EOL to ensure that schema tokens are not mixed in case files are not properly ended with EOL
+            String content = Utils.getFileContent(path) + "\n";
```

Comment:
Please use `System.lineSeparator()` instead of `"\n"`

Reply:
Good spot, will make this change.

This shouldn't matter much since the built string is simply fed to the GraphQL parser (we're not generating a file out of it), but maybe the parser does use platform-specific line endings, who knows 😄